DNN Platform/Library/Entities/Modules/ModuleController.cs

```diff
@@ -1902,6 +1902,18 @@ public void MoveModule(int moduleId, int fromTabId, int toTabId, string toPaneNa
             //Move the module to the Tab
             dataProvider.MoveTabModule(fromTabId, moduleId, toTabId, toPaneName, UserController.Instance.GetCurrentUserInfo().UserID);
 
+            //Update the Tab reference for the module's ContentItems
+            var contentController = Util.GetContentController();
+            var contentItems = contentController.GetContentItemsByModuleId(moduleId);
+            if (contentItems != null && contentItems.Count() > 0)
+            {
+                foreach (var item in contentItems)
+                {
+                    item.TabID = toTabId;

```

Comment:
This can be improved I think:

```
if (item.TabID != toTabId)
{
    item.TabID = toTabId;
    contentController.UpdateContentItem(item);
}
```